### PR TITLE
Bluetooth: Mesh: Convert k_work usage

### PR DESCRIPTION
**Pull request by trond-snekvik**
Convert all Bluetooth Mesh usage of the deprecated k_work API.

Signed-off-by: Trond Einar Snekvik <Trond.Einar.Snekvik@nordicsemi.no>

### CODE REVIEW

File: subsys/bluetooth/mesh/light_ctrl_srv.c [hunk 1]
```diff
@@ -94,6 +94,12 @@ static bool is_enabled(const struct bt_mesh_light_ctrl_srv *srv)
 	return srv->lightness->ctrl == srv;
 }
 
+static int32_t delay_remaining(struct bt_mesh_light_ctrl_srv *srv)
```
Comment (alxelax) [inline subsys/bluetooth/mesh/light_ctrl_srv.c:97]: Shouldn't it return `uint32_t` since `int32_t` causes implicit type conversion later in comparison and mathematic equations?

File: subsys/bluetooth/mesh/light_ctrl_srv.c [hunk 2]
```diff
@@ -173,22 +179,22 @@ static void light_set(struct bt_mesh_light_ctrl_srv *srv, uint16_t lvl,
...
 
-static uint32_t remaining_fade_time(struct bt_mesh_light_ctrl_srv *srv)
+static uint32_t curr_fade_time(struct bt_mesh_light_ctrl_srv *srv)
 {
 	if (!atomic_test_bit(&srv->flags, FLAG_TRANSITION)) {
 		return 0;
 	}
 
-	return k_delayed_work_remaining_get(&srv->timer);
+	return srv->fade.duration - remaining_fade_time(srv);
```
Comment (alxelax) [inline subsys/bluetooth/mesh/light_ctrl_srv.c:197]: Might it cause a negative value (as a result to the wrong unsigned value) due to the ceiling within `remaining_fade_time`?

File: subsys/bluetooth/mesh/light_ctrl_srv.c [hunk 3]
```diff
@@ -638,6 +643,10 @@ static void delayed_action_timeout(struct k_work *work)
 		.time = srv->fade.duration
 	};
 
+	if (!is_enabled(srv)) {
```
Comment (alxelax) [inline subsys/bluetooth/mesh/light_ctrl_srv.c:646]: It doesn't seem related to delayable work API changes :)

**Review by alxelax — CHANGES_REQUESTED** [review]

File: subsys/bluetooth/mesh/light_ctrl_srv.c (hunk 3 above)
Comment (trond-snekvik) [reply subsys/bluetooth/mesh/light_ctrl_srv.c:646]: It is. We can not assume cancel() succeeds, and need to exit early in case it failed

File: samples/bluetooth/mesh/light/src/model_handler.c [hunk 4]
```diff
@@ -113,7 +114,8 @@ static void led_work(struct k_work *work)
 /* Set up a repeating delayed work to blink the DK's LEDs when attention is
  * requested.
  */
-static struct k_delayed_work attention_blink_work;
+static struct k_work_delayable attention_blink_work;
+static bool attention;
```
[the remark this answers is not in the document]
Comment (trond-snekvik) [reply samples/bluetooth/mesh/light/src/model_handler.c:118]: This is not necessary here, as this is only being accessed once in each context, and only as a write in one and read in the other

**Review by mia-ko — APPROVED** [review]
:)

**Review by alxelax — APPROVED** [review]

**Review by Balaklaka — APPROVED** [review]
